src/plugins/types.ts

```diff
@@ -22,3 +37,64 @@ export const assertInt64: Plugin = {
     return undefined;
   },
 };
+
+export function floodBigints(
+  primitiveIntType0: "int64" | "int53" | IntegerType,
+  allowed: Partial<Record<OpCode | Assignment["kind"], "bigint" | "int">>,
+): Plugin {
+  const primitiveIntType = type(primitiveIntType0) as IntegerType;
+  return {
+    name: "floodBigints",
+    visit(node, spine) {
+      let nodeType: Type;
+      try {
+        nodeType = getType(isAssignment(node) ? node.variable : node, spine);
```

Comment:
When can this throw?

Reply:
Apparently never in tests. I copied it from `assertInt64` but this plugin is run before nodes whose type cannot be calculated are introduced whereas `assertInt64` is ran at the very end.